docs: complete overhaul and switch to `mkdocs` (#321)
fluxcd/helm-operator: docs/helmrelease-guide/automation.md

# Automation

This section of the guide is mostly a clarification of some common
misconceptions about (non-existent) automation features of the Helm Operator.

## Image updates

Because the Helm Operator is a [Flux umbrella project](https://fluxcd.io),
occasionally people assume it is capable of updating image references in
the `HelmRelease` and/or associated charts. This feature is however baked
in to [Flux](https://github.com/fluxcd/flux), and not the Helm Operator
itself due to it having no knowledge of available images or the origin of the
`HelmRelease`.

For more details about this Flux feature, please refer to the [documentation
for the `HelmRelease` integration](https://docs.fluxcd.io/en/stable/references/helm-operator-integration.html).

## Helm repository chart updates

Another much requested feature is automated updates for charts from [Helm
repository chart sources](chart-sources.md#helm-repositories). The
development of this feature is currently blocked until the automation logic

> has been untangled from Flux, to keep up-to-date about new developments of

Reviewer: ```suggestion
has been untangled from Flux. To keep up-to-date about new developments of
```

this feature you can subscribe to
[fluxcd/helm-operator#12](https://github.com/fluxcd/helm-operator/issues/12).

> It is possible to get a likewise functionality by making use of an

Reviewer: ```suggestion
It is possible to get similar functionality by making use of an
```

"likewise" is a great word -- I would deploy it like this: 

> This is how you frobnicate A:
> ...
> B can be done likewise, but with the frobs reversed.

[umbrella chart](https://helm.sh/docs/howto/charts_tips_and_tricks/#complex-charts-with-many-dependencies)
from a [Git repository chart source](chart-sources.md#git-repositories) with
a [version range set](https://helm.sh/docs/topics/chart_best_practices/dependencies/#versions),

> as for charts from Git repository sources a [dependency update is performed by

Reviewer: ```suggestion
as for charts from Git repository sources, a [dependency update is performed by
```

> default](chart-sources.md#dependency-updates) that will download the latest

Reviewer: ```suggestion
default](chart-sources.md#dependency-updates), and that will download the latest
```
